Mathlib/Topology/AlexandrovDiscrete.lean

```diff
@@ -51,3 +54,108 @@ lemma isClopen_sUnion (hS : ∀ s ∈ S, IsClopen s) : IsClopen (⋃₀ S) :=
+(interior_maximal (iInter_mono λ _ ↦ interior_subset) $ isOpen_iInter λ _ ↦
+  isOpen_interior).antisymm' $ subset_iInter λ _ ↦ interior_mono $ iInter_subset _ _
+
+lemma interior_sInter (S : Set (Set α)) : interior (⋂₀ S) = ⋂ s ∈ S, interior s := by
+  simp_rw [sInter_eq_biInter, interior_iInter]
+
+lemma closure_iUnion (f : ι → Set α) : closure (⋃ i, f i) = ⋃ i, closure (f i) :=
+compl_injective $ by simpa only [←interior_compl, compl_iUnion] using interior_iInter λ i ↦ (f i)ᶜ
+
+lemma closure_sUnion (S : Set (Set α)) : closure (⋃₀ S) = ⋃ s ∈ S, closure s := by
+  simp_rw [sUnion_eq_biUnion, closure_iUnion]
+
+end AlexandrovDiscrete
+
+variable {s t : Set α} {x y : α}
+
+/-- The *exterior* of a set is the intersection of all its neighborhoods. In an Alexandrov-discrete
+space, this is the smallest neighborhood of the set. -/
+def exterior (s : Set α) : Set α := (𝓝ˢ s).ker
+
+lemma exterior_singleton_eq_ker_nhds (a : α) : exterior {a} = (𝓝 a).ker := by simp [exterior]
+
+lemma exterior_def (s : Set α) : exterior s = ⋂₀ {t : Set α | IsOpen t ∧ s ⊆ t} := by
+  ext a
+  simp only [exterior, mem_ker, mem_sInter, mem_setOf_eq, and_imp, mem_nhdsSet_iff_forall]
+  refine' ⟨λ hs t ht hts ↦ hs _ λ b hb ↦ ht.mem_nhds $ hts hb, λ hs t ht ↦ interior_subset $
+    hs (interior t) isOpen_interior _⟩
+  simpa only [←mem_interior_iff_mem_nhds] using ht
+
+lemma subset_exterior_iff : s ⊆ exterior t ↔ ∀ U, IsOpen U → t ⊆ U → s ⊆ U := by
+  simp [exterior_def]
+
+lemma interior_subset_iff : interior s ⊆ t ↔ ∀ U, IsOpen U → U ⊆ s → U ⊆ t := by
+  simp [interior]
+
+lemma subset_exterior : s ⊆ exterior s := subset_exterior_iff.2 $ λ _ _ ↦ id
+
+lemma exterior_minimal (h₁ : s ⊆ t) (h₂ : IsOpen t) : exterior s ⊆ t := by
+  rw [exterior_def]; exact sInter_subset_of_mem ⟨h₂, h₁⟩
+
+lemma IsOpen.exterior_eq (h : IsOpen s) : exterior s = s :=
+(exterior_minimal Subset.rfl h).antisymm subset_exterior
+
+lemma IsOpen.exterior_subset_iff (ht : IsOpen t) : exterior s ⊆ t ↔ s ⊆ t :=
+⟨subset_exterior.trans, λ h ↦ exterior_minimal h ht⟩
+
+@[mono] lemma exterior_mono : Monotone (exterior : Set α → Set α) :=
+λ _s _t h ↦ ker_mono $ nhdsSet_mono h
+
+@[simp] lemma exterior_empty : exterior (∅ : Set α) = ∅ := isOpen_empty.exterior_eq
+@[simp] lemma exterior_univ : exterior (univ : Set α) = univ := isOpen_univ.exterior_eq
+
+@[simp] lemma exterior_eq_empty : exterior s = ∅ ↔ s = ∅ :=
+⟨eq_bot_mono subset_exterior, by rintro rfl; exact exterior_empty⟩
+
+variable [AlexandrovDiscrete α]
+
+@[simp] lemma isOpen_exterior : IsOpen (exterior s) := by
+  rw [exterior_def]; exact isOpen_sInter $ λ _ ↦ And.left
+
+@[simp] lemma exterior_eq_iff_isOpen : exterior s = s ↔ IsOpen s :=
+  ⟨λ h ↦ h ▸ isOpen_exterior, IsOpen.exterior_eq⟩
+
+@[simp] lemma exterior_subset_iff_isOpen : exterior s ⊆ s ↔ IsOpen s := by
+  simp only [exterior_eq_iff_isOpen.symm, Subset.antisymm_iff, subset_exterior, and_true]
+
+lemma exterior_subset : exterior s ⊆ t ↔ ∃ U, IsOpen U ∧ s ⊆ U ∧ U ⊆ t :=
```

Comment:
```suggestion
lemma exterior_subset_iff : exterior s ⊆ t ↔ ∃ U, IsOpen U ∧ s ⊆ U ∧ U ⊆ t :=
```

To match [subset_interior_iff](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/Basic.html#subset_interior_iff)